pool.go

```diff
@@ -497,21 +517,6 @@ func (p *Pool) doOverflowDrain() {
 }
 
 func (p *Pool) getExisting() (*ioErrConn, error) {
-	// Fast-path if the pool is not empty. Return error if pool has been closed.
```

Comment:
We need to keep this fast path here, we don't want to have to create/fetch a new timer on every command as that adds extra overhead.

You should be able to abstract that for loop into a new method though, which accepts a timer channel which can be nil, and then use that method in both places we're currently selecting on `p.pool` here.

PR looks great otherwise though, thanks for doing this!